nixos/modules/programs/environment.nix

```diff
@@ -55,14 +57,15 @@ in
          export TERM=$TERM
 
          unset ASPELL_CONF
-         for i in ${concatStringsSep " " (reverseList cfg.profiles)} ; do
+         for i in ${concatStringsSep " " (reverseList cfg.environment.profiles)} ; do
            if [ -d "$i/lib/aspell" ]; then
              export ASPELL_CONF="dict-dir $i/lib/aspell"
            fi
          done
 
          export NIX_USER_PROFILE_DIR="/nix/var/nix/profiles/per-user/$USER"
-         export NIX_PROFILES="${concatStringsSep " " (reverseList cfg.profiles)}"
+         export NIX_PROFILES="${concatStringsSep " " (reverseList cfg.environment.profiles)}"
+         ${optionalString cfg.nixup.enable "export NIXUP_CONFIG=\"$XDG_CONFIG_HOME/nixup/profile.nix\""}
```

Comment:
Likewise, this can simply be the default so we don't clutter the user environment